core/store/src/db.rs

```diff
@@ -39,6 +40,9 @@ pub(crate) enum DBOp {
     /// Sets `key` to `value`, and additionally debug-checks that the value is
     /// not overwritten.
     Insert { col: DBCol, key: Vec<u8>, value: Vec<u8> },
+    /// Merges `value` into `key`. The column must have a merge operator
+    /// defined.
+    MergeValue { col: DBCol, key: Vec<u8>, value: Vec<u8> },
```

Comment:
This is probably out of scope for this PR, but I feel we are starting to have many DBOp and different columns types here: insert-only columns which only support Insert, ref-counted columns supports UpdateRefCount, merge columns that support merge, and some of the column types are non-compatible with other column types. For example, a column shouldn't both support merge and rc, or a merge column should not be insert-only. I don't think it's too big a problem now because these special type columns are not that many, but it seems to become more complicated. Just writing down my thoughts here, probably we don't need to do anything in this PR.